Formula/gap.rb

```diff
@@ -0,0 +1,137 @@
+  sha256 "4c5a5bbbdb5275213cc575174f7bfb8bfd61c12bd082770798162b2ae6ac577d"
+
+  bottle do
+    cellar :any
+    sha256 "714ba2f943796f16be21d9dd7ac562873c481c38d8a47620f99b8571c19a25f7" => :high_sierra
+    sha256 "b745466b7b76c750b79d00f9ca6c47c154a0a2e18988caf4fcd8bd15c282697d" => :sierra
+    sha256 "fdaf283fb6c9657dc75e142b957759bc63c5b9a81f5545564df1b030ccb83564" => :el_capitan
+    sha256 "9eca000ecddafb1101edb06cd6087f87e38f9d168d5224e8c7030ccf6bf85853" => :x86_64_linux
+  end
+
+  # NOTE:  the archive contains the [GMP library](https://gmplib.org/) under
+  #   `extern/`, which is not needed if it is already installed (for example,
+  #   with Homebrew), and a number of GAP packages under `pkg/`, some of
+  #   which need to be built.
+
+  option "with-packages",
+         "Try to build included packages using BuildPackages script"
+
+  deprecated_option "with-InstPackages" => "with-packages"
+
+  depends_on "gmp"
+  # NOTE:  A version of [GMP](https://gmplib.org/) is included in GAP archive
+  #   under `extern/`, it is possible to use it instead of the brewed `gmp`.
+  #   See https://www.gap-system.org/Download/INSTALL for details.
+
+  depends_on "readline" => :recommended
+
+  def install
+    # Remove some useless files
+    rm Dir["bin/*.bat", "bin/*.ico", "bin/*.bmp", "bin/cygwin.ver"]
+
+    # Remove GMP archives (`gmp` formula is declared as a dependency)
+    rm Dir["extern/gmp-*.tar.gz"]
+
+    # XXX:  Currently there is no `install` target in `Makefile`.
+    #   According to the manual installation instructions in
+    #
+    #     https://www.gap-system.org/Download/INSTALL ,
+    #
+    #   the compiled "bundle" is intended to be used "as is," and there is
+    #   no instructions for how to remove the source and other unnecessary
+    #   files after compilation.  Moreover, the content of the
+    #   subdirectories with special names, such as `bin` and `lib`, is not
+    #   suitable for merging with the content of the corresponding
+    #   subdirectories of `/usr/local`.  The easiest temporary solution seems
+    #   to be to drop the compiled bundle into `<prefix>/libexec` and to
+    #   create a symlink `<prefix>/bin/gap` to the startup script.
+    #   This use of `libexec` seems to contradict Linux Filesystem Hierarchy
+    #   Standard, but is recommended in Homebrew's "Formula Cookbook."
+
+    args = %W[--prefix=#{libexec} --with-gmp=system]
+
+    args << "--#{build.with?("readline") ? "with" : "without"}-readline"
+    # NOTE: `--with-readline` is the default, it is included for clarity
+
+    system "./configure", *args
+
+    # Fix a bug caused by the buggy `configure` which does not respect
+    # `--prefix` when generating the startup script: the variable `GAP_DIR`
+    # is being set to the (temporary) build directory, but should be set to
+    # the value of `--prefix` option.
+    ["bin/gap-default32.sh", "bin/gap-default64.sh"].each do |startup_script|
+      next unless File.exist?(startup_script)
```

Comment:
We should know which files exist, and only replace in those.

Reply:
Which file is created depends on whether a 32bit or 64bit build is made.  Since I see no provisions in this brew file to force a 32bit build, and since modern Macs all are running 64bit CPUs, you might want to hardcode `bin/gap-default64.sh`. If, however, you expect people on 32bit system to use this system, you'd have to add a check for this... All in all, though, I don't see what is gained over the above very simple command?